### DISCUSSION TITLE
是否有可以直接在MaaFramework中使用的繁體中文OCR模型？

### DISCUSSION THREAD

**Discussion by UnexIstiD** [MaaXYZ/MaaFramework#311] [Q&A] [answered]
由於本人最近正在開發的遊戲僅有支援繁體中文，而paddleOCR的提供者雖然有提供繁體中文的模型，但是我找不到對應的模型下載處
![螢幕擷取畫面 2024-08-10 205204](https://github.com/user-attachments/assets/c9716a7b-c85f-4fe5-a081-ae01467df0ff)
因此想詢問一下，是否有大佬已有訓練出來的模型？（最好是可以直接放置在/resource/model的onnx檔案與對應的key.txt檔案）

**Comment by moomiji** [comment]
可直接给 maa 使用的 OCR 模型资源在这：https://github.com/MaaXYZ/MaaCommonAssets/tree/main/OCR

**Comment by MistEO** [comment]
https://github.com/MaaXYZ/MaaCommonAssets/tree/main/OCR%2Fppocr_v3%2Fzh_tw

**Reply by UnexIstiD** [reply]
感謝回復，目前繁體是可以用的，但是這模型辨識的準確率有些差強人意，如下方案例
![螢幕擷取畫面 2024-08-11 012432](https://github.com/user-attachments/assets/307d8a03-fbbc-43ac-a542-21bf9b726600)

Case 1
「征」辨識為「在」
「：（全形冒號）」辨識為「主」
![螢幕擷取畫面 2024-08-11 012347](https://github.com/user-attachments/assets/0df86fcb-302a-4a32-953e-574427390cae)

Case 2
「啟」辨識為「廠」
![螢幕擷取畫面 2024-08-11 012627](https://github.com/user-attachments/assets/0f8f9c6c-9588-4dfd-bbb0-1fff307f463e)

Case 3
「:（半形冒號）」辨識為「!（半形驚嘆號）」
![螢幕擷取畫面 2024-08-11 012732](https://github.com/user-attachments/assets/f313a2cc-fdd4-4e93-8db0-f10a7696d3c6)
想請問下是否有其他的繁中模型可以試試看？或者我可以如何進行再訓練？

**Reply by MistEO** [reply]
使用 Paddle OCR 官方模型进行 finetuning即可，具体参考他们官方的文档，我们提供的这个就是官方的模型

可以参考的资料：[隔壁明日方舟 MAA 的训练方法](https://github.com/MaaAssistantArknights/MaaAI/blob/main/common/OCR/README.md)

**Reply by MistEO** [reply]
训练完成后再参考 [这个命令](https://github.com/MaaXYZ/MaaCommonAssets/blob/main/OCR/README.md) 转换成 onnx

**Reply by MistEO** [reply]
一些小的识别错误也可以直接通过 pipeline 的 replace 字段处理

**Comment by MistEO** [comment, marked as answer]
可以试试 v5 了

https://github.com/MaaXYZ/MaaFramework/issues/651
